File: Difficulty: Easy/Count Digits/count-digits.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std;

// } Driver Code Ends
class Solution{
public:
    int evenlyDivides(int N){
        //code here
        int cnt=0, temp=N;
        while(N!=0){
            int k=N%10;
            if(k!=0 && temp%k==0){
                cnt++;
            }
            N/=10;
        }
        return cnt;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N;
        cin>>N;
        Solution ob;
        cout << ob.evenlyDivides(N) << endl;
    
cout << "~" << "\n";
}
    return 0; 
}
// } Driver Code Ends

File: Difficulty: Easy/Count Digits/count-digits.py
#User function Template for python3

class Solution:
    def evenlyDivides(self, n):
        num = n
        count = 0
        while n:
            x = n%10
            if x:
                count = count + int(num%x==0)
            n = n//10
        return count
        
#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())

        ob = Solution()
        print(ob.evenlyDivides(N))
        print("~")

# } Driver Code Ends